scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJobFactory.java

```diff
@@ -143,14 +144,26 @@ private static InternalJob createJob(TaskFlowJob userJob) throws JobCreationExce
         // keep an initial job content
         job.setTaskFlowJob(userJob);
         Map<Task, InternalTask> tasksList = new LinkedHashMap<>();
-        boolean hasPreciousResult = false;
+
+        ExecutorService pool = Executors.newWorkStealingPool(8);
 
         for (Task t : userJob.getTasks()) {
-            tasksList.put(t, createTask(userJob, job, t));
+            pool.execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        tasksList.put(t, createTask(userJob, job, t));
+                    } catch (JobCreationException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+        }
 
-            if (!hasPreciousResult) {
-                hasPreciousResult = t.isPreciousResult();
-            }
+        pool.shutdown();
+        try {
+            pool.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
+        } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00108)